src/test/spec/unified_runner/operation.rs

```diff
@@ -164,13 +191,19 @@ pub(crate) struct Operation {
 }
 
 impl Operation {
-    pub(crate) async fn execute<'a>(&self, test_runner: &TestRunner, description: &str) {
+    pub(crate) async fn execute(&self, test_runner: &TestRunner, description: &str) {
+        let _ = self.execute_fallible(test_runner, description).await;
+    }
+
+    async fn execute_fallible(&self, test_runner: &TestRunner, description: &str) -> Result<()> {
```

Comment:
The test `withTransaction` operation needs to have any error from the child operations propagated, independently of their `expectError` fields, so this makes that available.